bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -252,7 +253,7 @@ void ReplicaImp::validatedMessageHandler(CarrierMesssage *msg) {
     }
   }
   if (!isCollectingState()) {
-    onMessage<T>(std::make_unique<T>(trueTypeObj));
+    onMessage<T>(std::unique_ptr<T>(trueTypeObj));
```

Comment:
nit: just carious why did you change this all over the code?
In general, std::make_unique is safer during compile time (wil lgenerate more errors if you did something wrong).

Reply:
make_unique calls a copy constructor and creates a new object of type "T", which causes a memory leak.